src/Data/Rational/Unnormalised/Properties.agda

```diff
@@ -514,6 +933,93 @@ p ≤? q = Dec.map′ *≤* drop-*≤* (↥ p ℤ.* ↧ q ℤ.≤? ↥ q ℤ.* 
 *-distrib-+ : _DistributesOver_ _≃_ _*_ _+_
 *-distrib-+ = *-distribˡ-+ , *-distribʳ-+
 
+------------------------------------------------------------------------
+-- Properties of _*_ and -_
+
+neg-distribˡ-* : ∀ p q → - (p * q) ≃ - p * q
+neg-distribˡ-* p q = *≡* $ cong (ℤ._* (↧ p ℤ.* ↧ q))
+                         $ ℤ.neg-distribˡ-* (↥ p) (↥ q)
+
+neg-distribʳ-* : ∀ p q → - (p * q) ≃ p * - q
+neg-distribʳ-* p q = *≡* $ cong (ℤ._* (↧ p ℤ.* ↧ q))
+                         $ ℤ.neg-distribʳ-* (↥ p) (↥ q)
+------------------------------------------------------------------------
+-- Properties of _*_ and _≤_
+
+private
+  reorder₁ : ∀ a b c d → a ℤ.* b ℤ.* (c ℤ.* d) ≡ a ℤ.* c ℤ.* b ℤ.* d
+  reorder₁ = solve 4 (λ a b c d → a :* b :* (c :* d) := a :* c :* b :* d) refl
+    where open ℤ-solver
+
+  reorder₂ : ∀ a b c d → a ℤ.* b ℤ.* (c ℤ.* d) ≡ a ℤ.* c ℤ.* (b ℤ.* d)
+  reorder₂ = solve 4 (λ a b c d → a :* b :* (c :* d) := a :* c :* (b :* d)) refl
+    where open ℤ-solver
+
+*-cancelʳ-≤-pos : ∀ {r} → Positive r → ∀ {p q} → p * r ≤ q * r → p ≤ q
+*-cancelʳ-≤-pos {mkℚᵘ +[1+ n ] dm} _ {p} {q} (*≤* x≤y)
+  = let o = dm ℕ.+ n ℕ.* suc dm ; l₁ = ↥ p ℤ.* ↧ q ; l₂ = ↥ q ℤ.* ↧ p
+  in *≤* $ ℤ.*-cancelʳ-≤-pos l₁ l₂ o $ begin
+  l₁ ℤ.* (+[1+ n ] ℤ.* +[1+ dm ])          ≡⟨ reorder₂ (↥ p) _ _ (ℤ.+ (suc dm)) ⟩
+  ↥ p ℤ.* +[1+ n ] ℤ.* (↧ q ℤ.* +[1+ dm ]) ≤⟨ x≤y ⟩
+  ↥ q ℤ.* +[1+ n ] ℤ.* (↧ p ℤ.* +[1+ dm ]) ≡⟨ reorder₂ (↥ q) _ _ (ℤ.+ (suc dm)) ⟩
+  l₂ ℤ.* (+[1+ n ] ℤ.* +[1+ dm ])          ∎ where open ℤ.≤-Reasoning
+
+*-cancelˡ-≤-pos : ∀ {r} → Positive r → ∀ {p q} → r * p ≤ r * q → p ≤ q
+*-cancelˡ-≤-pos {r} r>0 {p} {q}
+  rewrite *-comm-≡ r p
+        | *-comm-≡ r q = *-cancelʳ-≤-pos r>0
+
+*-monoˡ-≤-non-neg : ∀ {r} → NonNegative r → (_* r) Preserves _≤_ ⟶ _≤_
```

Comment:
If we do decide to use them this suffix should be `nonNeg` rather than `non-neg`

Reply:
Re: this comment and the one above, I tried being consistent with how these properties are named in `Data.Integer.Properties`. However these properties do not seem to use the predicates you introduced, and directly construct positive (or negative) integers from implicit arguments.

The way I see it, there are two ways to go about it:
- Rewrite theses properties in `Data.Integer.Properties` to use `Positive/...` predicates, but keep the naming.
- Rename here to what you're suggesting: `nonNeg` etc.

So it's really a matter of consistency, and I have no idea which path to follow.

Reply:
Or we could remove the `Positive/Negative` predicates from these, and directly construct positive & negative rationals as is done in `Data.Integer.Properties`. Though he type signatures will suffer.

Reply:
> However these properties do not seem to use the predicates you introduced

Yup, it's on the road map for when we switch over to using these predicates as instance arguments everywhere in v2.0. I think I'd also be keen to rename them to match my suggestion here to maintain consistency. Unlike switching to using predicates, this can be done in v1.4 via deprecation warnings.

Reply:
> I think I'd also be keen to rename them to match my suggestion here to maintain consistency
Perfect! I'll rename like you suggested and call it a day.